### ISSUE TITLE
Updating log pi function of Tanh Distribution

### ISSUE THREAD

**Issue by avnishn** [rlworkgroup/garage#1174]
Supposedly, using the following log pi calculation is more numerically stable than using the one in the original sac implementation:

    logp_pi -= tf.reduce_sum(2*(np.log(2) - pi - tf.nn.softplus(-2*pi)), axis=1)

**Comment by avnishn** [comment]
https://github.com/openai/spinningup/blob/038665d62d569055401d91856abb287263096178/spinup/algos/tf1/sac/core.py#L54
